x-pack/plugins/lens/public/persistence/lens_store.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectAttributes } from 'src/legacy/server/saved_objects';
+
+export interface LensDocument {
+  id?: string;
+  type?: string;
+  visualizationType: string | null;
+  datasourceType: string | null;
+  title: string;
+  lensState: {
+    datasource: unknown;
+    visualization: unknown;
+  };
+}
+
+interface SavedObectStore {
+  create: (type: string, object: SavedObjectAttributes) => Promise<{ id: string }>;
+  update: (type: string, id: string, object: SavedObjectAttributes) => Promise<{ id: string }>;
+  get: (
+    type: string,
+    id: string
+  ) => Promise<{ id: string; type: string; attributes: SavedObjectAttributes }>;
+}
+
+const DOC_TYPE = 'lens';
+
+export interface LensDocumentSaver {
```

Comment:
Same thing about the naming here - I don't think we need to prefix all the names with lens